components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java

```diff
@@ -1933,7 +2135,25 @@ public IdentityProvider getIDPbyId(Connection dbConnection, int idpId, int tenan
     public IdentityProvider getIDPbyResourceId(Connection dbConnection, String resourceId, int tenantId,
                                        String tenantDomain) throws IdentityProviderManagementException {
 
-        return getIDP(dbConnection, null, -1, resourceId, tenantId, tenantDomain);
+        return getIDP(dbConnection, null, -1, resourceId, tenantId, tenantDomain, true);
+    }
+
+    /**
+     * Retrieves an identity provider.
+     *
+     * @param dbConnection    Database connection.
+     * @param resourceId      UUID of the identity provider.
+     * @param tenantId        Tenant ID of the identity provider.
+     * @param tenantDomain    Tenant domain of the identity provider.
+     * @param plainTextSecret If true, the secrets of the federated authenticators will be retrieved in plain text.
+     * @return An identity provider.
+     * @throws IdentityProviderManagementException Error while retrieving the identity provider.
+     */
+    public IdentityProvider getIDPbyResourceId(Connection dbConnection, String resourceId, int tenantId,
```

Comment:
can we deprecate the old methods?

Reply:
deprecated the old method with 6a2f91760f8d0931e1bc54964973f61ba589d79a